app/actions/SurveyActions.js

```diff
@@ -0,0 +1,164 @@
+// @flow
+
+import { Survey } from './ActionTypes';
+import callAPI from 'app/actions/callAPI';
+import { surveySchema } from 'app/reducers';
+import type { Thunk } from 'app/types';
+import { startSubmit, stopSubmit } from 'redux-form';
+import { push } from 'react-router-redux';
+import { addNotification } from 'app/actions/NotificationActions';
+
+export function fetchAll() {
+  return callAPI({
+    types: Survey.FETCH,
+    endpoint: '/surveys/',
+    schema: [surveySchema],
+    meta: {
+      errorMessage: 'Henting av spørreundersøkelser feilet',
+      queryString: ''
+    },
+    propagateError: true
+  });
+}
+
+export function fetch(surveyId: number): Thunk<*> {
+  return dispatch =>
+    dispatch(
+      callAPI({
+        types: Survey.FETCH,
+        endpoint: `/surveys/${surveyId}/`,
+        schema: surveySchema,
+        meta: {
+          errorMessage: 'Henting av spørreundersøkelse feilet'
+        },
+        propagateError: true
+      })
+    );
+}
+
+export function addSurvey(data: Object): Thunk<*> {
+  return dispatch => {
+    dispatch(startSubmit('survey'));
+    return dispatch(
+      callAPI({
+        types: Survey.ADD,
+        endpoint: '/surveys/',
+        method: 'POST',
+        body: data,
+        schema: surveySchema,
+        meta: {
+          errorMessage: 'Legg til spørreundersøkelse feilet'
+        }
+      })
+    )
+      .then(action => {
+        if (!action || !action.payload) return;
+        const id = action.payload.result;
+        dispatch(stopSubmit('survey'));
+        dispatch(addNotification({ message: 'Spørreundersøkelse lagt til.' }));
+        dispatch(push(`/surveys/${id}`));
```

Comment:
Would it not be better to navigate in the components, rather than in the actions to the api?